### PR TITLE
Context: Fix CreateThread partial initialization issue

### PR DESCRIPTION
**Pull request by skmp**
#### Overview
`Context::CreateThread` would add a thread to the `Threads` vector before it has been fully constructed / initialized.

### CODE REVIEW

File: External/FEXCore/Source/Interface/Core/Core.cpp [hunk 1]
```diff
@@ -553,14 +550,7 @@ namespace FEXCore::Context {
   }
 
   FEXCore::Core::InternalThreadState* Context::CreateThread(FEXCore::Core::CPUState *NewThreadState, uint64_t ParentTID) {
-    FEXCore::Core::InternalThreadState *Thread{};
-
-    // Grab the new thread object
-    {
-      std::lock_guard<std::mutex> lk(ThreadCreationMutex);
-      Thread = Threads.emplace_back(new FEXCore::Core::InternalThreadState{});
-      Thread->ThreadManager.TID = ++ThreadID;
```
Comment (neobrain) [inline External/FEXCore/Source/Interface/Core/Core.cpp:562]: Why is there no equivalent to this line in the new code?

Comment (skmp) [reply External/FEXCore/Source/Interface/Core/Core.cpp:562]: This TID is incorrect, and seems like some leftover from when FEX implementing its own TID and PIDs. `ThreadID` isn't even an atomic.

The correct value is set by `Context::InitializeThreadTLSData` called from `Context::ExecutionThread`, which is run with the host backing thread created from `Context::InitializeThread`.

Comment (skmp) [reply External/FEXCore/Source/Interface/Core/Core.cpp:562]: (The actual thread creation is done using a custom threading implementaiton in `FEXCore/Source/Utils/Thread.cpp `class PThread`)

**Review by Sonicadvance1 — APPROVED** [review]
Looks good, cleaning up old cruft.

**Comment by skmp** [comment]
Obliterated `ThreadID` from the context since not used, updated docstrings, will merge if passes.

**Pull request merged**